pkg/manifest/build.go

```diff
@@ -22,6 +22,11 @@ type Build struct {
 	dependents   []Pipeline
 	repos        []rpmmd.RepoConfig
 	packageSpecs []rpmmd.PackageSpec
+
+	// TODO: make private?
```

Comment:
Would love to get ideas/feedback on this - it seems a pattern we use in `manifest.Build` but not too sure if it's appropriate here?

Reply:
The pattern/rule for these things is that private members should be options that are mandatory and shouldn't be changed outside the constructor.  So we make them private and force them to be set by making them an argument to the `NewX()` function.  Anything optional, options that aren't absolutely required for a pipeline or image, are public and are set by any caller that needs them.

Reply:
Thanks for your input on this! This sounds like my approach was okay(?). I did push a small commit for review that uses a slightly different route by introducing a new optional "BuildOptions" that can be omited. But happy to remove it again, I realize it probably does not fit that well. Just let me know what you prefer :) (or if there is a third option that I might have overlooked).

Reply:
It could be a good solution for extending the options in the future.  Might be worth considering the same for other pipelines at some point.